tests/multi_ptr/multi_ptr_common_constructors.h

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for multi_ptr common constructors
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_COMMON_CONSTRUCTORS_H
+#define __SYCLCTS_TESTS_MULTI_PTR_COMMON_CONSTRUCTORS_H
+
+#include "../common/common.h"
+#include "../common/get_cts_string.h"
+#include "../common/type_list.h"
+
+namespace multi_ptr_common_constructors {
+
+template <typename T, sycl::access::address_space Space,
+          sycl::access::decorated Decorated>
+struct multi_ptr_kernel_name;
+
+/** @brief Provdes text description of test case in case of fail
+ *  @tparam Space sycl::access::address_space value
+ *  @tparam Decorated sycl::access::decorated value
+ */
+template <sycl::access::address_space Space, sycl::access::decorated Decorated>
+std::string get_case_description(const std::string &info, size_t overload_index,
+                                 const std::string &type_name) {
+  static std::vector<std::string> overloads{
+      "multi_ptr()", "multi_ptr(const multi_ptr&)", "multi_ptr(multi_ptr&&)",
+      "multi_ptr(multi_ptr<ElementType, Space, yes>::pointer)",
+      "multi_ptr(std::nullptr_t)"};
+  std::string addr_space{sycl_cts::get_cts_string::for_address_space<Space>()};
+  std::string decor{sycl_cts::get_cts_string::for_decorated<Decorated>()};
+  std::string message{info + " of get() for " + overloads[overload_index] +
+                      " with tparams: <" + addr_space + "> <" + decor +
+                      "> and type: " + type_name};
+  return message;
+}
+
+template <sycl::access::address_space Space>
+constexpr bool use_local_accessor =
+    (Space == sycl::access::address_space::local_space) ||
+    (Space == sycl::access::address_space::private_space);
+
+/** @brief Provdes verification of multi_ptr common constructors with template
+ *         parameters given
+ *  @tparam T Variable type for type coverage
+ *  @tparam Space sycl::access::address_space value
+ *  @tparam Decorated sycl::access::decorated value
+ */
+template <typename T, sycl::access::address_space Space,
+          sycl::access::decorated Decorated>
+void run_tests(sycl_cts::util::logger &log, const std::string &type_name) {
+  using namespace sycl_cts;
+  using m_ptr_t = sycl::multi_ptr<T, Space, Decorated>;
+  using other_decorated_ptr_t =
+      sycl::multi_ptr<T, Space, sycl::access::decorated::yes>;
+
+  // Kernel name
+  using k_name = multi_ptr_kernel_name<T, Space, Decorated>;
+
+  constexpr size_t types_size = 5;
```

Comment:
Add a comment as to why this is 5. Same for below.

Reply:
I agree. Comment added.